### PR TITLE
#156882168 User should be able to view game log of past games

### PR DESCRIPTION
**Pull request by amarachukwu-agbo**
#### What does this PR do?

Lets user view a game log of past games

#### Description of Task to be completed?
 - Add API endpoint to get games history
- Add API endpoint to get leaderboard
- Add API endpoint to get donations
- Design user interface to view the game log

#### How should this be manually tested?

Clone the repo and install application dependencies. Run command  `npm start` and open the application on the browser. Click on the dashboard link on the index page and you should be redirected to the dashboard where you will see the gamelog, donations and leaderboard.

#### Any background context you want to provide?

N/A

#### What are the relevant pivotal tracker stories?

#156882168

### CODE REVIEW

**Review by Spectrumsun — COMMENTED** [review]
Well done. Everything works well.

**Review by kenoseni — COMMENTED** [review]
Good work Amara for implementing this feature, everything worked well. The only thing you should add to the dashboard is the ability for a user to sign out.

File: app/helpers/index.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+ */
+export default class Helper {
+  /**
+       * Group Games by their winners
+       * @param {Array} gamesArray - All games played
+       * @param {String} winner - The game winner
+       * @returns {Object} - The grouped leaderboard
+       */
+  static groupByWinner(gamesArray, winner) {
+    const grouped = gamesArray.reduce((acc, obj) => {
```
Comment (andela-moe) [inline app/helpers/index.js]: Abbreviations `acc, obj` are not good variable or parameter names. Please use something more descriptive in this case. Also avoid using dataTypes `obj`, `gamesArray ` as variable/parameters names

Comment (andela-moe) [reply app/helpers/index.js]: Please fix this and other instances of this in this entire PR

**Review by andela-moe — COMMENTED** [review]
Good job with your test.  Please review the comments and resolve.

File: app/helpers/index.js (hunk 1 above)
Comment (amarachukwu-agbo) [reply app/helpers/index.js]: Fixed

**Pull request merged**